pkgs/development/mobile/androidenv/compose-android-packages.nix

```diff
@@ -2,7 +2,7 @@
 , licenseAccepted ? false
 }:
 
-{ toolsVersion ? "26.1.1"
+{ cmdLineToolsVersion ? "8.0"
 , platformToolsVersion ? "33.0.3"
 , buildToolsVersions ? [ "33.0.1" ]
 , includeEmulator ? false
```

Comment:
Can we still include a `toolsVersion ? null` after cmdLineToolsVersion and assert that it remains null? (And tell people to either remove it or switch to `cmdLinetoolsVersion` if it's provided?)

Reply:
it's obsolete, and if you check the repo.json there's only one version of it available. However, if you download the `cmdline-tools` independently, then extract it, then try to download any packages with `sdkmanager`, then it always downloads and extracts the `tools`! That's why I didn't give any choice to people! Apparently it should always be there,even that it's obsolete. If you like I can check it twice.

Reply:
Oh, I'm not suggesting we still support it. I'm suggesting that we assert that people _don't_ try to provide it, and help them out if they do.

Reply:
Try using a boolean instead of a nullable. check [952737f](https://github.com/NixOS/nixpkgs/pull/208793/commits/952737ff484a22eed6069771a05ffc3ae44a9236). Makes sense?

Reply:
I think you are misunderstanding. I also think we should fully drop `tools` as it is obsolete. But if a user provides it in their shell environment we want to tell them to upgrade using an assert failure.

Reply:
>it's obsolete, and if you check the repo.json there's only one version of it available. However, if you download the cmdline-tools independently, then extract it, then try to download any packages with sdkmanager, then it always downloads and extracts the tools! That's why I didn't give any choice to people! Apparently it should always be there,even that it's obsolete. If you like I can check it again.

Maybe I'm misunderstanding too? It always requires the tools?

Reply:
These are the facts that can impact our decision here:
- `tools` is obsolete, but `sdkmanager` always download it in all SDKs.
- There are some old binaries in `tools` that are not available on any other packages, like `uiautomatorviewer`.
- There's only one version of `tools` available in `repo.json`.
- In the `repository2-1.xml`, which is the source for `repo.json`, you can find that `built-tools` are depending on `tools`. Notice `build-tools` is mandatory in the SDK. Maybe that's why `sdkmanager` keeps the `tools` in all SDKs.
```xml
  <remotePackage path="build-tools;30.0.3">
    <!--Generated from bid:6966805, branch:rvc-qpr1-release-->
    <type-details xsi:type="generic:genericDetailsType"/>
    <revision>
      <major>30</major>
      <minor>0</minor>
      <micro>3</micro>
    </revision>
    <display-name>Android SDK Build-Tools 30.0.3</display-name>
    <uses-license ref="android-sdk-license"/>
    <dependencies>
      <dependency path="tools"/>
    </dependencies>
```

so there's no choice for the version of `tools`, and that's why I choose a boolean instead of a nullable. Maybe it's a good idea in the future to read the `dependencies` and build up the packages tree, without asking people to fill out the dependencies by hand! Let me know what do you think.

Reply:
Also I just noticed
```xml
  <remotePackage obsolete="true" path="tools">
    <!--Generated from:ab bid:4333796 branch:git_studio-sdk-release-->
    <type-details xsi:type="generic:genericDetailsType"/>
    <revision>
      <major>26</major>
      <minor>1</minor>
      <micro>1</micro>
    </revision>
    <display-name>Android SDK Tools</display-name>
    <uses-license ref="android-sdk-license"/>
    <dependencies>
      <dependency path="patcher;v4"/>
      <dependency path="emulator"/>
      <dependency path="platform-tools">
        <min-revision>
          <major>20</major>
        </min-revision>
```
which means `tools` is depending on the `patcher`, `emulator`, and `platform-tools`. Let me create a commit to make them visible to the `tools`.

Reply:
I specified those dependencies in [9495595](https://github.com/NixOS/nixpkgs/pull/208793/commits/9495595145ec54bac03dfe08ced5ee8319ae41f3)

Reply:
>  Maybe it's a good idea in the future to read the dependencies and build up the packages tree, without asking people to fill out the dependencies by hand! Let me know what do you think.

This is a good idea!

Reply:
Alright, I see now. That's confusing. I would still specify toolsVersion in case there _is_ an update at some point then, and because it makes composeAndroidPackages compatible with existing code.